### PR TITLE
#65 Compatibility with Symfony 3

### PR DESCRIPTION
**Pull request by JHGitty**
Thanks to @mkrauser (https://github.com/mkrauser/phone-number-bundle/commit/1d71d935b9c84c597723742b9b6d856e0bf2dd65)

### CODE REVIEW

**Comment by JHGitty** [comment]
@mkrauser @xabbuh @thewilkybarkid 
This issue contains Travis CI build errors. Could anybody please fix this? Thanks :+1:

**Comment by thewilkybarkid** [comment]
@JHGitty I fixed the tests in #67.

**Comment by thewilkybarkid** [comment]
Ok, so a couple of things to fix:

Could you update `PhoneNumberTypeTest` to be something like:

```php
if (method_exists('Symfony\\Component\\Form\\FormTypeInterface', 'getName')) {
    $type = new PhoneNumberType();
} else {
    $type = 'Misd\\PhoneNumberBundle\\Form\\Type\\PhoneNumberType';
}
```

and the `jms/serializer-bundle` constraint needs to be `~0.11|~1.0` otherwise `symfony/framework-bundle` is stuck on `2.8.x`.

**Comment by thewilkybarkid** [comment]
We also need to increase the memory limit on PHP 5.3 builds, https://github.com/javiereguiluz/EasyAdminBundle/pull/346 looks like it does the trick.

**Comment by JHGitty** [comment]
I don't really understand how to fix ``PhoneNumberTypeTest``. Could you provide a fully working fix for it?

File: Tests/Doctrine/DBAL/Types/PhoneNumberTypeTest.php [hunk 1]
```diff
@@ -41,7 +41,13 @@ class PhoneNumberTypeTest extends TestCase
 
     public static function setUpBeforeClass()
     {
-        Type::addType('phone_number', 'Misd\PhoneNumberBundle\Doctrine\DBAL\Types\PhoneNumberType');
+        if (method_exists('Symfony\\Component\\Form\\FormTypeInterface', 'getName')) {
+            $type = new PhoneNumberType();
+        } else {
+            $type = 'Misd\\PhoneNumberBundle\\Form\\Type\\PhoneNumberType';
+        }
```
Comment (xabbuh) [inline Tests/Doctrine/DBAL/Types/PhoneNumberTypeTest.php]: Not sure what you are trying to do here, but this is a test for a Doctrine type and not for a form type.

Comment (JHGitty) [inline Tests/Doctrine/DBAL/Types/PhoneNumberTypeTest.php]: Ah, I was really confused and tired while doing this. Thanks. :smile:

Comment (thewilkybarkid) [inline Tests/Doctrine/DBAL/Types/PhoneNumberTypeTest.php]: @JHGitty I meant `Misd\PhoneNumberBundle\Tests\Form\Type\PhoneNumberTypeTest`, not `Misd\PhoneNumberBundle\Tests\Doctrine\DBAL\Types\PhoneNumberTypeTest`.

Comment (JHGitty) [inline Tests/Doctrine/DBAL/Types/PhoneNumberTypeTest.php]: Fixed.

**Comment by JHGitty** [comment]
@thewilkybarkid Everything is fine :+1:

File: composer.json [hunk 2]
```diff
@@ -17,16 +17,16 @@
...
         "giggsey/libphonenumber-for-php": "~5.7|~6.0|~7.0",
-        "symfony/framework-bundle": "~2.1"
+        "symfony/framework-bundle": "~2.1|~3.0"
     },
     "require-dev": {
         "doctrine/doctrine-bundle": "~1.0",
-        "jms/serializer-bundle": "~0.11",
+        "jms/serializer-bundle": "~0.11|~1.0",
         "phpunit/phpunit": "~4.0",
         "symfony/form": "~2.3",
```
Comment (thewilkybarkid) [inline composer.json]: Missed one :)

Comment (JHGitty) [inline composer.json]: Fixed. :smile:

**Comment by JHGitty** [comment]
@thewilkybarkid The 5.3 build fails again, I don't understand why. I've changed the memory_limit to 1280MB and 1500MB but it does not work.

Allowed memory size of 1073741824 bytes exhausted (tried to allocate 71 bytes)

**Comment by JHGitty** [comment]
@thewilkybarkid The PHP 5.3 build is failing again. I don't understand why.

**Comment by thewilkybarkid** [comment]
Thanks. I can't release v1.1.0 at the moment, so I've committed as a941972 and released v1.0.6 instead.

**Comment by JHGitty** [comment]
@thewilkybarkid Shouldn't this also be merged into master?

**Comment by thewilkybarkid** [comment]
Yeah, that's coming.

**Pull request closed without merging**